### PR TITLE
Fixing #795: find_homography_dlt_iterated sometimes fails

### PR DESCRIPTION
**Pull request by dkoguciuk**
### Description
The solution to #795 -- all discussion can be found in the issue thread.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [x] Did you discuss the functionality or any breaking changes before ?
- [x] **Pass all tests**: did you test in local ? `make test`
- [x] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [x] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

**Comment by dkoguciuk** [comment]
> LGTM once cuda test passes

It looks like it fails:

>   failed to register layer: Error processing tar file(exit status 1): write /usr/local/lib/python3.7/dist-packages/sklearn/linear_model/_stochastic_gradient.py: no space left on device
>   Error: Docker pull failed with exit code 1

**Comment by edgarriba** [comment]
@dkoguciuk seems a problem with the CI. We will check with @Borda

**Comment by edgarriba** [comment]
@dkoguciuk seems that macOS doesn’t like to create multiple seeds
https://github.com/kornia/kornia/runs/2644514405?check_suite_focus=true

**Comment by dkoguciuk** [comment]
@edgarriba , the problem is not macOS specific -- it is present for pytorch <= `v1.7.0`

There was a pytorch [bug](https://github.com/pytorch/pytorch/issues/33546) where `torch.manual_seed()` and `torch.initial_seed()` functions had incompatible types. Pytorch v1.6.0 (published on `Jul 25, 2020`) doesn't have this bugfix (fixed on `Aug 12, 2020`). 

What do you suggest now? It is an obvious pytorch bug, should we bypass it somehow?

**Comment by edgarriba** [comment]
a quick one is to skip the test if only if `pytorch.__version__ <= 1.7.0`

**Comment by dkoguciuk** [comment]
Ok, let me code it!

**Comment by dkoguciuk** [comment]
@edgarriba , the fix is proposed here: https://github.com/kornia/kornia/pull/1032
